operator/internal/controller/vectorized/cluster_controller_test.go

```diff
@@ -0,0 +1,386 @@
+package vectorized
+
+import (
+	"context"
+	"fmt"
+	"slices"
+	"strings"
+	"testing"
+	"time"
+
+	certmanagerv1 "github.com/cert-manager/cert-manager/pkg/apis/certmanager/v1"
+	"github.com/cockroachdb/errors"
+	"github.com/go-logr/logr/testr"
+	"github.com/redpanda-data/common-go/rpadmin"
+	"github.com/redpanda-data/redpanda-operator/pkg/testutil"
+	"github.com/stretchr/testify/require"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8slabels "k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/utils/ptr"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/metrics/server"
+
+	redpanda "github.com/redpanda-data/redpanda-operator/charts/redpanda/v5/client"
+	vectorizedv1alpha1 "github.com/redpanda-data/redpanda-operator/operator/api/vectorized/v1alpha1"
+	crds "github.com/redpanda-data/redpanda-operator/operator/config/crd/bases"
+	"github.com/redpanda-data/redpanda-operator/operator/pkg/admin"
+	"github.com/redpanda-data/redpanda-operator/operator/pkg/labels"
+	"github.com/redpanda-data/redpanda-operator/operator/pkg/resources"
+	resourcetypes "github.com/redpanda-data/redpanda-operator/operator/pkg/resources/types"
+	"github.com/redpanda-data/redpanda-operator/operator/pkg/secrets"
+	"github.com/redpanda-data/redpanda-operator/pkg/kube"
+	"github.com/redpanda-data/redpanda-operator/pkg/kube/kubetest"
+	"github.com/redpanda-data/redpanda-operator/pkg/otelutil/log"
+)
+
+func TestClusterControllerGolden(t *testing.T) {
+	tests := []struct {
+	name           string
+	clusterDef     func() *vectorizedv1alpha1.Cluster
+	reconcilerFunc func(t *testing.T, mgr manager.Manager) *ClusterReconciler
+	goldenFile     string
+	normalizerFunc func(ctx context.Context, testScheme *runtime.Scheme, cluster *vectorizedv1alpha1.Cluster, t *testing.T, ctl *kube.Ctl) []client.Object
+}{
+		{
+			name:           "minimal_cluster",
+			clusterDef:     minimalClusterDef,
+			reconcilerFunc: minimalReconciler,
+			normalizerFunc: minimalClusterNormalizer,
```

Comment:
Neither of these should be customizable on a per test case basis. You lose the value of having a table driven tests as any changes here effectively create completely different test cases all together.